dotnet/src/VectorDataIntegrationTests/VectorDataIntegrationTests/DependencyInjectionTests.cs

```diff
@@ -0,0 +1,214 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System.Linq.Expressions;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.VectorData;
+using Xunit;
+
+namespace VectorDataSpecificationTests;
+
+public abstract class DependencyInjectionTests<TVectorStore, TCollection, TKey, TRecord>
+    where TVectorStore : VectorStore
+    where TCollection : VectorStoreCollection<TKey, TRecord>
+    where TKey : notnull
+    where TRecord : class
+{
+    protected abstract void PopulateConfiguration(ConfigurationManager configuration, object? serviceKey = null);
+
+    protected abstract void RegisterVectorStore(IServiceCollection services, ServiceLifetime lifetime, object? serviceKey = null);
+
+    protected abstract void RegisterCollection(IServiceCollection services, ServiceLifetime lifetime, string collectionName = "name", object? serviceKey = null);
+
+    [Fact]
```

Comment:
Better to use `[ConditionalFact]` everywhere in case we need to skip for some reason.

Reply:
For now I would prefer not to do it, as these tests should just work (example: throwing `ArgumentNullException` should always just work no matter if we can connect to a db/service or not). Once I implement tests for other connectors and find a test that requires it, then I am going to add it.